ntoskrnl/mm/ARM3/mdlsup.c

```diff
@@ -1672,18 +1672,36 @@ MmProtectMdlSystemAddress(IN PMDL MemoryDescriptorList,
 }
 
 /*
- * @unimplemented
+ * @implemented
  */
+_IRQL_requires_max_(APC_LEVEL)
 VOID
 NTAPI
-MmProbeAndLockProcessPages(IN OUT PMDL MemoryDescriptorList,
-                           IN PEPROCESS Process,
-                           IN KPROCESSOR_MODE AccessMode,
-                           IN LOCK_OPERATION Operation)
+MmProbeAndLockProcessPages(_Inout_ PMDL MemoryDescriptorList,
+                           _In_ PEPROCESS Process,
+                           _In_ KPROCESSOR_MODE AccessMode,
+                           _In_ LOCK_OPERATION Operation)
 {
-    UNIMPLEMENTED;
-}
+    KAPC_STATE ApcState;
+    BOOLEAN IsAttached = FALSE;
+
+    if (Process != PsGetCurrentProcess())
+    {
+        KeStackAttachProcess(&Process->Pcb, &ApcState);
+        IsAttached = TRUE;
+    }
 
+    _SEH2_TRY
```

Comment:
```suggestion
    /* Protect in try/finally to ensure we detach even if MmProbeAndLockPages() throws an exception */
    _SEH2_TRY
```
Note that the thrown exception, if any, would then be rethrown to the caller of MmProbeAndLockPages(), as it should.

Reply:
Added the comment.